accounts/views.py

```
@@ -31,3 +35,74 @@ def form_valid(self, form):
+            allowed_hosts={request.get_host()},
+        ):
+            return super().get_success_url()
+
+        # If absolute URL and current request is secure, only allow https scheme
+        if is_absolute and request.is_secure() and parsed.scheme != "https":
+            return super().get_success_url()
+
+        # Determine canonical admin paths to avoid redirecting users there
+        try:
+            wagtail_admin_home = reverse("wagtailadmin_home")
+        except Exception:
```

Ruff (0.12.2)

Do not catch blind exception: `Exception`

(BLE001)